src/runtime/importhook.cs

```diff
@@ -115,187 +97,154 @@ internal static void SaveRuntimeData(RuntimeDataStorage storage)
 
         internal static void RestoreRuntimeData(RuntimeDataStorage storage)
         {
-            InitImport();
             storage.GetValue("py_clr_module", out py_clr_module);
             var rootHandle = storage.GetValue<IntPtr>("root");
             root = (CLRModule)ManagedType.GetManagedObject(rootHandle);
+            BorrowedReference dict = Runtime.PyImport_GetModuleDict();
+            Runtime.PyDict_SetItemString(dict.DangerousGetAddress(), "clr", py_clr_module);
```

Comment:
I'd add a property `ClrModuleReference` that returns a `BorrowedReference` of `py_clr_module`, and then used it in `PyDict_SetItemString` overload, that works with `BorrowedReferences` to avoid using `Dangerous*` methods.